packages/grafana-ui/src/types/plugin.ts

```diff
@@ -76,16 +83,38 @@ export interface PluginMetaInfo {
 }
 
 export class AppPlugin {
-  components: {
+  meta: PluginMeta;
+
+  angular?: {
     ConfigCtrl?: any;
+    pages: { [component: string]: any };
   };
 
-  pages: { [str: string]: any };
-
-  constructor(ConfigCtrl: any) {
-    this.components = {
-      ConfigCtrl: ConfigCtrl,
+  constructor(meta: PluginMeta, pluginExports: any) {
```

Comment:
problem with this is that we want the constructor to be (eventually when apps start using this) used by the plugin code (module.ts) so sending in meta wont work then, has to be done later by the app via some setter. 

I was thinking something like:
```
export cont plugin = new AppPlugin()
  .setAngularConfig(ConfigCtrl)
  .setAngularPage(MyPageController);
```

Something we can refactor to after this PR, just wondering if you had other plans

Reply:
+1 for using the same form as panel plugins.

This is a bit more weird because the meta defines what needs to be imported.  My goal was to get the apps working again with this PR, and then improve it in #16586